src/sass/parse.js

```diff
@@ -4177,34 +4180,68 @@ function getSC() {
 }
 
 /**
- * Check if token is part of a hexadecimal number (e.g. `#fff`) inside
- *      a simple selector
+ * Check if token is part of a hexadecimal number (e.g. `#fff`) inside a simple
+ * selector
  * @param {number} i Token's index number
  * @return {number}
  */
 function checkShash(i) {
-  var l;
+  const start = i;
+  let l;
 
-  if (i >= tokensLength || tokens[i].type !== TokenType.NumberSign) return 0;
+  if (i >= tokensLength) return 0;
+
+  if (tokens[i].type === TokenType.NumberSign) {
+    if (checkInterpolation(i)) return 0;
+    i++;
+  } else return 0;
+
+  while (i < tokensLength) {
+    if (l = checkIdentOrInterpolation(i) || checkNumber(i)) i += l;
+    else if (tokens[i].type === TokenType.HyphenMinus) i += 1;
```

Comment:
Great! This will be very helpful!

I can't think of any exceptions off the top of my head, so on first thought putting it in `checkIdentOrInterpolation` would make perfect sense.